sql/core/src/main/scala/org/apache/spark/sql/api/python/PythonSQLUtils.scala

```diff
@@ -72,6 +72,14 @@ private[sql] object PythonSQLUtils extends Logging {
     FunctionRegistry.functionSet.flatMap(f => FunctionRegistry.builtin.lookupFunction(f)).toArray
   }
 
+  def listAllSQLConfigsWithTags():
+    Array[(String, String, String, String, Array[String])] = {
+    val conf = new SQLConf()
+    conf.getAllDefinedConfsWithTags.map(c =>
+      Tuple5(c._1, c._2, c._3, c._4, c._5.toArray)
+    ).toArray
+  }
+
   private def listAllSQLConfigs(): Seq[(String, String, String, String)] = {
```

Comment:
We can remove some of these other methods if we go with this PR's approach of assigning arbitrary tags to configs.

That includes:
- `listAllSQLConfigs`
- `listRuntimeSQLConfigs`
- `listStaticSQLConfigs`